### PR TITLE
Delta Lake diff: Upgrade delta-rs crate version

### PR DESCRIPTION
**Pull request by Jonathan-Rosenberg**
## Change Description

To resolve [this](https://github.com/treeverse/lakeFS/security/dependabot/103) security advisory, this PR updates the version of the `deltalake` crate to the latest one and also bumps the openssl crate version to 0.10.48.

## How was this tested?

Manually using the Web UI

### CODE REVIEW

File: pkg/plugins/diff/delta_diff_server/src/utils.rs [hunk 1]
```diff
@@ -31,48 +30,39 @@ const TIMESTAMP_KEY: &str = "timestamp";
 
 pub(crate) fn construct_storage_config(config: GatewayConfig) -> HashMap<String, String> {
     let mut s3_config: HashMap<String, String> = HashMap::new();
-    s3_config.insert("AWS_ACCESS_KEY_ID".to_string(), config.key);
-    s3_config.insert("AWS_ENDPOINT_URL".to_string(), config.endpoint);
-    s3_config.insert("AWS_S3_ADDRESSING_STYLE".to_string(), "path".to_string());
-    s3_config.insert("AWS_SECRET_ACCESS_KEY".to_string(), config.secret);
-    s3_config.insert("AWS_STORAGE_ALLOW_HTTP".to_string(), "true".to_string());
+    s3_config.insert(s3_storage_options::AWS_ACCESS_KEY_ID.to_string(), config.key);
```
Comment (arielshaqed) [inline pkg/plugins/diff/delta_diff_server/src/utils.rs:33]: I'd expect these options to be usable directly somehow, it's really strange to have a non-string type that cannot be used.  Can we make `s3_config` accept a `s3_storage_option` or whatever that type is?

File: pkg/plugins/diff/delta_diff_server/src/utils.rs [hunk 2]
```diff
@@ -31,48 +30,39 @@ const TIMESTAMP_KEY: &str = "timestamp";
 
 pub(crate) fn construct_storage_config(config: GatewayConfig) -> HashMap<String, String> {
     let mut s3_config: HashMap<String, String> = HashMap::new();
```
Comment (arielshaqed) [inline pkg/plugins/diff/delta_diff_server/src/utils.rs:32]: Not in this PR, but it seems odd to have a mutable map for a static map.  Can't we just set up an immutable map here?

**Review by arielshaqed — APPROVED** [review]
Thanks!

How was this tested?  {Shoul,Di}d we also test this manually on the GUI?

File: pkg/plugins/diff/delta_diff_server/src/utils.rs (hunk 2 above)
Comment (Jonathan-Rosenberg) [reply pkg/plugins/diff/delta_diff_server/src/utils.rs:32]: changed it
thanks!

**Pull request merged**